accepted/2020/diagnostics/activity-improvements-2.md

```diff
@@ -0,0 +1,165 @@
+# Improve Activity API usability and OpenTelemetry integration (Part 2)
+
+We have exposed some [Activity APIs to improve the API usability and to integrate with OpenTelemetry](https://github.com/dotnet/designs/pull/98) (OT) so Activity can be a suitable replacement for the OT Span type described by the [OT Specs](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#span). We have received feedback regarding some OT blocked scenarios which need to be addressed in the Activity APIs and unblock the OT SDK implementation.
+
+This document is listing the proposed changes and the rationale for the proposal. **As we still receiving more feedback and there are some issues under discussion, it is expected this documents will get updated before we do the full design review for the proposed changes.**
+
+## ActivityContext.IsRemote
+
+We already exposed the ActivityContext type as described by the OT specs. There is a property inside this type called `IsRemote` which we didn't include in the first place because it was under discussion with the OT spec committee. It has been confirmed this property is needed and will stay as part of the OT specs.
+
+[ActivityContext.IsRemote](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#spancontext) is a Boolean flag which returns true if the ActivityContext was propagated from a remote parent. When creating children from remote spans, their IsRemote flag MUST be set to false.
+
+```c#
+    public readonly partial struct ActivityContext : IEquatable<ActivityContext>
+    {
+        // This constructor already exist but we are adding extra defaulted parameter for isRemote.
+        public ActivityContext(ActivityTraceId traceId, ActivitySpanId spanId, ActivityTraceFlags traceFlags, string? traceState = null, isRomte = false) {...}
+    
+        // This is the new property 
+        public bool IsRemote { get; }
+    }
+```
+
+Activity class will need to expose such property too to be able to set it as part of the included context. This is needed by the OT asp.net and Http client OT adapters.
+
+```c#
+    public partial class Activity : IDisposable
+    {
+        public bool IsRemote { get; set; }
```

Comment:
It will be good to describe the case that it needs to be set after creation usually speeds up the discussion, same for the Kind setter below.

Reply:
good idea.